SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -259,12 +259,19 @@ Worker_RequestId USpatialSender::CreateEntity(USpatialActorChannel* Channel)
 
 Worker_ComponentData USpatialSender::CreateLevelComponentData(AActor* Actor)
 {
-	if (FLevelData* LevelData = ClassInfoManager->SchemaDatabase->LevelPathToLevelData.Find(NetDriver->World->GetName()))
+	UWorld* ActorWorld = Actor->GetTypedOuter<UWorld>();
+	if (ActorWorld != NetDriver->World)
 	{
-		if (uint32* ComponentId = LevelData->SublevelNameToComponentId.Find(Actor->GetTypedOuter<UWorld>()->GetName()))
+		FString CleanLevelPath = UWorld::RemovePIEPrefix(ActorWorld->GetOuter()->GetPathName());
+		uint32* ComponentId = ClassInfoManager->SchemaDatabase->LevelPathToComponentId.Find(CleanLevelPath);
+		if (ComponentId != nullptr)
 		{
 			return ComponentFactory::CreateEmptyComponentData(*ComponentId);
 		}
+		else
+		{
+			UE_LOG(LogSpatialSender, Warning, TEXT("Could not find Streaming Level Component for Level %s. Have you generated schema?"), *CleanLevelPath);
```

Comment:
Can you put the Actor's name here as well? Also isn't this more of an error than a warning?

Reply:
Done